src/eradius_server.erl

```diff
@@ -391,10 +387,45 @@ inc_reply_counter(discnak, NasProp) ->
 inc_reply_counter(_Cmd, _NasProp) ->
     ok.
 
+inc_request_counter_accounting(accreq = Cmd, NasProp, Ms, #radius_request{attrs = Attrs}) ->
+    AccReqs = build_counter_accounting(Cmd, Attrs, [{accounting, accountRequests}]),
+    lists:foreach(fun({Account, AccountRequest}) ->
+        eradius_metrics:update_nas_request(Account, NasProp#nas_prop.metrics_info, Ms),
+        eradius_counter:inc_request_counter(AccountRequest, NasProp)
+    end, AccReqs);
+inc_request_counter_accounting(_, _, _, _) ->
+    ok.
```

Comment:
I don't see why this has to use a list (but I might be missing something here).

If you leave the inc_{reply,request}_counter for accreq/resp in place, you would only have to deal with the Status-Type here and you could use a plain match on the Status-Type.
Also, calling inc_request_counter_accounting/inc_response_counter_accounting from inc_request_counter(accreq, _)/inc_reply_counter(accresp, _) might further simplify the logic.

Reply:
Done.